packages/contracts/contracts/L2/predeploys/TssRewardContract.sol

```diff
@@ -0,0 +1,144 @@
+    uint256 public totalAmount;
+
+    // set call address
+    constructor(address _deadAddress, address payable _owner) {
+        deadAddress = _deadAddress;
+        owner = _owner;
+    }
+
+    /**
+     * Enforces that the modified function is only callable by a specific null address.
+     *  authenticated to call this function.
+     */
+    modifier onlyFromDeadAddress() {
+        require(
+            msg.sender == deadAddress,
+            "tss reward call message unauthenticated"
+        );
+        _;
+    }
+
+    modifier onlyOwner() {
+        require(
+            msg.sender == owner,
+            "only be called by the owner of this contract"
+        );
+        _;
+    }
+
+    modifier checkBalance() {
+        require(
+            address(this).balance >= totalAmount,
+            "balance record and contract balance are not equal"
+        );
+        _;
+    }
+
+    /**
+     * @dev return the total undistributed amount
+     */
+    function queryReward() external view checkBalance returns (uint256) {
+        return address(this).balance;
+    }
+
+    /**
+     * @dev claimReward distribute reward to tss member.
+     * @param _blockStartHeight The block height at L2 which needs to distribute profits
+     * @param _length The distribute batch block number
+     * @param _tssMembers The address array of tss group members
+     */
+    function claimReward(uint256 _blockStartHeight, uint32 _length, address[] calldata _tssMembers)
+        external
+        virtual
+        onlyFromDeadAddress
+        checkBalance
+    {
+        uint256 blockAmount = ledger[_blockStartHeight];
+        uint256 sendAmount = 0;
+        uint256 accu = 0;
+        // release reward from _blockStartHeight to _blockStartHeight + _length
+        for (uint256 i=_blockStartHeight; i<_length; i++) {
+            // calc send amount
+            sendAmount = blockAmount.div(_tssMembers.length);
+            // delete distributed height
+            delete ledger[_blockStartHeight];
+            for (uint256 j=0; j < _tssMembers.length; j++) {
+                address payable addr = payable(_tssMembers[j]);
+                accu = accu.add(sendAmount);
+                totalAmount = totalAmount.sub(sendAmount);
+                addr.transfer(sendAmount); // TODO use call to transfer
+            }
+            uint256 reserved = blockAmount.sub(accu);
+            require(reserved >= 0, "release amount gt real balance");
+            if (reserved > 0) {
+                dust = dust.add(reserved);
+            }
+        }
+
+        emit DistributeTssReward(
+            _blockStartHeight,
+            _length,
+            _tssMembers
+        );
+    }
+
+    /**
+     * @dev update tss member gas reward by every block.
+     * @param _blockID The block height at L2 which needs to distribute profits
+     * @return _tssMembers Address array of tss group members
+     */
+    function updateReward(uint256 _blockID, uint256 _amount)
```

Comment:
Update reward each block? updating reward each epoch looks better. cc @curryxbo @FletcherMan

Reply:
L2Geth does not know the actual batch definition.